apps/nextra/pages/en/build/smart-contracts/maps.mdx

```diff
@@ -0,0 +1,167 @@
+---
+title: "Maps"
+---
+
+# Overview
+
+There are multiple different implementations of key-value maps inside the framework, suited for different usecases.
+We will go over their differences and similarities, and how to choose which one to use.
+
+## Aptos Blockchain performance and gas cost considerations 
+
+State on the Aptos Blockchain is managed as a set of resources. Transactions
+performance heavily depends on how reads and writes to resources.
+Storage gas costs are paid based on number of resources that exist, and their sizes. 
+IO gas costs are paid based on number of resources read and modified, and their sizes, 
+but are generally significantly smaller than storage gas costs. 
+That means that writing to a new resource has the highest (storage) gas cost, and deleting
+an existing resource gives the largest refund. 
+Additionally, transactions modifying the same resource conflict with one another, and cannot be
+executed in parallel.
+
+One useful analogy is thinking about each resource being a file on a disk, 
+then performance of smart contract would correlate well to a program that 
+operates on files in the same way.
+
+## Different Map implementations
+
+- `OrderedMap` is a struct, and is, similar to `vector`, fully contained within the resource that stores it.
+  With it, it is bounded in size to the size of a single resource.
+  It provides regular map functions, as well as accessing elements in order, like front/back or prev/next. 
+  When you need an inline mapping, that will fit in a resource, this is the option to choose.
+  It's implementation is SortedVectorMap, but because of limited size and efficiency of memcpy, all main operations are practically O(log(n)). 
+- `Table` is unbounded in size, puts each (key, value) pair in the separate resource. You can `add` or `remove` elements, 
+  or check if it `contains` some key, but cannot be iterated on. When keys or values are large / unbounded, we can use the `Table`. 
+  Also if we want to parallelize transactions and we have a few elements that are modified extremely often, `Table` can provide that.
+  Note that `Table` cannot be destroyed, because it doesn't know if it is empty.
+  - `TableWithLength` is wrapper around the `Table`, that adds tracking of it's `length`, allowing `length`, `empty` and `destroy_empty`
+    operations on top of the `Table`. Adding or removing elements to `TableWithLength` cannot be done in parallel.
```

Comment:
highlight this "cannot"

shall we discourage people to use TableWithLength?

Reply:
you cannot use Table if you want to be able to destroy it. 

we probably need TableWithLength variant that uses aggregators. not sure if it is worth copying for that :) maybe TableWithParLength ? :)